easybuild/tools/run.py

```diff
@@ -589,3 +589,61 @@ def parse_log_for_error(txt, regExp=None, stdout=True, msg=None):
                   (regExp, '\n'.join([x[0] for x in res])))
 
     return res
+
+
+def extract_errors_from_log(log_txt, reg_exps):
+    """
+    Check log_txt for messages matching regExps and return warnings and errors
+    :param log_txt: String containing the log, will be split into individual lines
+    :param reg_exps: List of: regular expressions (as strings) to error on,
+                    or tuple of regular expression and action (any of [IGNORE, WARN, ERROR])
+    :return (warnings, errors) as lists of lines containing a match
+    """
+
+    # promote single string value to list, since code below expects a list
+    if isinstance(reg_exps, string_type):
+        reg_exps = [reg_exps]
+
+    re_tuples = []
+    for cur in reg_exps:
+        try:
+            if isinstance(cur, str):
+                reg_exp, action = cur, ERROR
+            else:
+                reg_exp, action = cur
+            if not isinstance(reg_exp, str) or action not in (IGNORE, WARN, ERROR):
+                raise TypeError("Invalid types")
```

Comment:
This should provide more details: invalid types for what, which types were expected, which were obtained

Reply:
This exception is caught 2 lines below and enhanced with more information including what you requested. But I enhanced it anyway a bit to distinguish the 2 failures